compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp

```diff
@@ -823,6 +905,77 @@ class ConvertNCHWToNHWC final : public luci::CircleNodeMutableVisitor<bool>
     post_trans->a(node);
     return true;
   }
+
+  bool visit(luci::CircleSub *node)
+  {
+    luci::CircleNode *pred_node = nullptr;
+    luci::CircleConst *subtract = nullptr;
+
+    if (is_NCHW_with_const_x(node, pred_node, subtract))
+    {
+      // case of subtract - pred_node
+      auto pre_trans = create_pre_transpose(node);
+      pre_trans->a(pred_node);
+
+      auto nhwc_const = create_NHWC_from_NCHW(subtract);
+      if (nhwc_const == nullptr)
+        return false;
+
+      node->x(nhwc_const);
+      node->y(pre_trans);
+    }
+    else if (is_NCHW_with_const_y(node, pred_node, subtract))
+    {
+      // case of pred_node - subtract
+      auto pre_trans = create_pre_transpose(node);
+      pre_trans->a(pred_node);
+
+      auto nhwc_const = create_NHWC_from_NCHW(subtract);
+      if (nhwc_const == nullptr)
+        return false;
+
+      node->x(pre_trans);
+      node->y(nhwc_const);
+    }
+    else
+    {
+      auto const_x = dynamic_cast<luci::CircleConst *>(node->x());
+      auto const_y = dynamic_cast<luci::CircleConst *>(node->y());
```

Comment:
This pattern appears three times in this PR. Actually, this information is used in all three cases.

I think the if statements can be restructured as follows.

```cpp
auto const_x = dynamic_cast<luci::CircleConst *>(node->x());
auto const_y = dynamic_cast<luci::CircleConst *>(node->y());

if (const_x == nullptr && const_y != nullptr)
{
  // Check NCHW for const_y
  // Insert pre-transpose
}
else if (const_x == nullptr && const_y != nullptr)
{
  // Check NCHW for const_x
  // Insert pre-transpose

}
else if (const_x == nullptr && const_y != nullptr)
{
  as-is
}
```

`post_NCHW_with_const` may be used for checking NCHW.

Reply:
Thanks for the suggestion!